pkg/dataobj/internal/util/slicegrow.go

```diff
@@ -0,0 +1,24 @@
+package slicegrow
+
+import "slices"
+
+// GrowToCap grows the slice to at least n elements total capacity.
+// It is an alternative to slices.Grow that increases the capacity of the slice instead of allowing n new appends.
+// This is useful when the slice is expected to have nil values.
+func GrowToCap[Slice ~[]E, E any](s Slice, n int) Slice {
```

Comment:
I found myself using this calculation in quite a few places so I extracted it. The regular call without subtracting the current len would double the slice on every iteration since we tend to have len==cap and assign directly in the readers